pajbot/bot.py

```diff
@@ -882,7 +882,7 @@ def parse_message(self, message, source, event, tags={}, whisper=False) -> bool:
             # once they are a founder they are always be a founder, regardless if they are a sub or not.
             if not source.founder:
                 source.founder = "founder" in badges
-            source.vip = "vip" in badges
+            source.vip = "vip" in (badges or tags)
```

Comment:
Is this some kind of fancy python generator syntax? Because otherwise wouldnt this first evaluate `badges or tags`, which is always `True`? Have you tested that with this change people without VIP status correctly get `vip=False` assigned?

Reply:
I would think it would have to be

```suggestion
            source.vip = "vip" in badges or "vip" in tags
```

Reply:
> Is this some kind of fancy python generator syntax? Because otherwise wouldnt this first evaluate `badges or tags`, which is always `True`? Have you tested that with this change people without VIP status correctly get `vip=False` assigned?

Yes, the part I missed in my screenshot was me testing my own status. See below, you're free to test (or pajlada probably will), but it seemingly works just fine:

![image](https://user-images.githubusercontent.com/12804673/193400167-9b0a0d44-3ce8-487b-8ad8-28c1b4ef9399.png)

Reply:
based on my very rough testing, it may be a case of python taking the first value in the brackets (in this case, badges):

```py
number = "56"
sentence = "xd"

print("56" in (number or sentence))
```
> True

```py
number = "56"
sentence = "xd"

print("56" in (sentence or number))
```
> False

Reply:
Subsequent question would be: can we use something like `"vip" in [badges,tags]`?